core/model/job/step/gotolocation/storage.go

```diff
@@ -0,0 +1,69 @@
+package gotolocation
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/openmarketplaceengine/openmarketplaceengine/redis/client"
+	"github.com/openmarketplaceengine/openmarketplaceengine/redis/marshalutils"
+)
+
+type Storage struct {
+	client     *redis.Client
+	expiration time.Duration
+}
+
+var storage = newStorage(3 * time.Hour)
+
+func newStorage(expiration time.Duration) *Storage {
+	return &Storage{
+		client:     client.NewStoreClient(),
+		expiration: expiration,
+	}
+}
+
+func key(driverID string) string {
+	return fmt.Sprintf("gotolocation-%s", driverID)
+}
+
+func (s *Storage) Store(ctx context.Context, goToLocation GoToLocation) error {
+	m, err := marshalutils.StructToMap(goToLocation)
+
+	if err != nil {
+		return fmt.Errorf("store StructToMap error: %w", err)
+	}
+
+	key := key(goToLocation.DriverID)
+
+	err = s.client.HSet(ctx, key, m).Err()
+	if err != nil {
+		return fmt.Errorf("store HSet error: %w", err)
+	}
+
+	s.client.Expire(ctx, key, s.expiration)
+
+	return nil
+}
+
+func (s *Storage) Retrieve(ctx context.Context, driverID string) (goToLocation GoToLocation, err error) {
```

Comment:
I'm not sure about the `driverID` being used as a lookup key here. Is it WIP?

Reply:
its a WIP, we are going to have some sort of job/step-execution-id.